src/frontend/lib/utils.ts

```diff
@@ -320,3 +320,22 @@ export function isEditingScreen(
 
   return !!EDITING_SCREEN_NAMES.find(val => val === routeName);
 }
+
+export function shallowEqual(a: any, b: any) {
+  console.log('in function');
+  if (a === b) return true;
+  if (
+    a === null ||
+    b === null ||
+    typeof a !== 'object' ||
+    typeof b !== 'object'
+  )
+    return false;
+  for (const key in a) {
+    if (!(key in b) || a[key] !== b[key]) return false;
+  }
+  for (const key in b) {
+    if (!(key in a)) return false;
+  }
+  return true;
+}
```

Comment:
is this from some existing implementation? if not, would lean towards using a well-tested module instead